show/main.py

```diff
@@ -1804,6 +1804,716 @@ def tablelize(keys, data, enable_table_keys, prefix):
+            healthpre = (cycle - avgerase) * 100.0 / cycle
+            click.echo("Device Health Status Is: {0:.2f}%".format(healthpre))
+        else:
+            click.echo("Can't get Average_Erase_Count attributes")
+
+    echo_empty_line()
+
+    #"""Check the later bad block status which may created"""
+    checkin = 0
+    testname = "SSD Badblock Test"
+    print_test_title(testname)
+    for line in outputattr:
+        if "Later_Bad_Block" in line:
+            rawval = line.split()[-1]
+            click.echo("Later_Bad_Block:         {0}".format(rawval))
+            checkin += 1
+
+        if "Later_Bad_Blk_Inf_R/W/E" in line:
+            rawval_read = line.split()[-3]
+            click.echo("Later_Bad_Blk_Inf Read:  {0}".format(rawval))
+            rawval_write = line.split()[-2]
+            click.echo("Later_Bad_Blk_Inf Write: {0}".format(rawval))
+            rawval_erase = line.split()[-1]
+            click.echo("Later_Bad_Blk_Inf Erase: {0}".format(rawval))
+            checkin += 1
+
+    if (checkin != 2):
+        click.echo("Can't get all 'Later_Bad_Block' attributes")
+
+    echo_empty_line()
+
+    #"""Read SSD temperature range 0~70 C"""
+    checkin = 0
+    testname = "SSD Temperature Test"
+    print_test_title(testname)
+    for line in outputattr:
+        if "Temperature_Celsius" in line:
+            rawval = line.split()[9]
+            temperature = int(rawval)
+            checkin = 1
+
+    if(checkin == 1):
+        click.echo("Temperature_Celsius: {0}C".format(temperature))
+    else:
+        click.echo("Can't get 'Temperature_Celsius' attributes")
+
+    echo_empty_line()
+
+
+#
+# 'ssd-help' command ####
+#
+
+@ssd.command()
+def help():
+    """show all ssd cmd"""
+
+    click.echo("\nUsage: show [options] [device]")
+    click.echo("========================= SHOW INFORMATION OPTIONS =========================\n")
+    click.echo("{0:30s}{1:<40}".format("    ssd firmwareinfo /dev/xxx",	"show SSD firmware info"))
+    click.echo("{0:30s}{1:<40}".format("    ssd capacity /dev/xxx     ",	"show SSD capacity"))
+    click.echo("{0:30s}{1:<40}".format("    ssd sn /dev/xxx          ",	"show SSD serial number"))
+    click.echo("{0:30s}{1:<40}".format("    ssd remaintime /dev/xxx  ",	"show SSD remaining time"))
+    click.echo("{0:30s}{1:<40}".format("    ssd pecycle /dev/xxx     ",	"show SSD P/E cycle"))
+    click.echo("{0:30s}{1:<40}".format("    ssd health /dev/xxx      ",	"show SSD health status which should be > 95% after MFG test"))
+    click.echo("{0:30s}{1:<40}".format("    ssd badblock /dev/xxx    ",	"show SSD bad block status which may created during test"))
+    click.echo("{0:30s}{1:<40}".format("    ssd temperature /dev/xxx ",	"read SSD temperature; range 0~70 C"))
+    click.echo("{0:30s}{1:<40}".format("    ssd all /dev/xxx         ",	"show all test iterms of SSD "))
+    click.echo("{0:30s}{1:<40}".format("    ssd help /dev/xxx        ",	"list help menu \n"))
+
+
+#
+# check_pcie_speed
```

Comment:
It would be better to have pcie in a separate PR
or at least fix the subject...